code/controllers/subsystem/discord.dm

```diff
@@ -113,3 +113,11 @@ SUBSYSTEM_DEF(discord)
 /datum/controller/subsystem/discord/proc/id_clean(input)
 	var/regex/num_only = regex("\[^0-9\]", "g")
 	return num_only.Replace(input, "")
+
+/datum/controller/subsystem/discord/proc/grant_role(id)
+	// Ignore this shit if config isnt enabled for it
+	if(!CONFIG_GET(flag/enable_discord_autorole))
+		return
+
+	// Make the request
+	rustg_http_request_blocking(RUSTG_HTTP_METHOD_PUT, "https://discordapp.com/api/guilds/[CONFIG_GET(string/discord_guildid)]/members/[id]/roles/[CONFIG_GET(string/discord_roleid)]", "",  "{\"Authorization\":\"Bot [CONFIG_GET(string/discord_token)]\"}")

```

Comment:
maybe add the request and response log here and put it into a separate log?

Suspect we'll probably need that at least once while configuring this.